File: python_merchant/examples/accounts/termsofservices/v1beta/retrieve_latest_terms_of_service_sample.py
# -*- coding: utf-8 -*-
# Copyright 2024 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""Module for retrieving the latest TermsOfService."""


# [START retrieveLatestTermsOfService]
from examples.authentication import generate_user_credentials
from google.shopping.merchant_accounts_v1beta import RetrieveLatestTermsOfServiceRequest
from google.shopping.merchant_accounts_v1beta import TermsOfServiceKind
from google.shopping.merchant_accounts_v1beta import TermsOfServiceServiceClient

# Replace with your actual values.
_REGION_CODE = "US"  # Replace with your region code
_KIND = (
    TermsOfServiceKind.MERCHANT_CENTER
)  # Replace with your TermsOfServiceKind


def retrieve_latest_terms_of_service():
  """Retrieves the latest TermsOfService for a specific region and kind."""

  credentials = generate_user_credentials.main()
  client = TermsOfServiceServiceClient(credentials=credentials)

  request = RetrieveLatestTermsOfServiceRequest(
      region_code=_REGION_CODE, kind=_KIND
  )

  try:
    print("Sending Retrieve Latest TermsOfService request:")
    response = client.retrieve_latest_terms_of_service(request=request)
    print("Retrieved TermsOfService below")
    print(response)
  except RuntimeError as e:
    print(e)


# [END retrieveLatestTermsOfService]

if __name__ == "__main__":
  retrieve_latest_terms_of_service()

File: php/examples/accounts/termsofservices/v1/RetrieveLatestTermsOfServiceSample.php
<?php
/**
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
require_once __DIR__ . '/../../../../vendor/autoload.php';
require_once __DIR__ . '/../../../Authentication/Authentication.php';
require_once __DIR__ . '/../../../Authentication/Config.php';
// [START merchantapi_retrieve_latest_termsofservice]
use Google\ApiCore\ApiException;
use Google\Shopping\Merchant\Accounts\V1\Client\TermsOfServiceServiceClient;
use Google\Shopping\Merchant\Accounts\V1\RetrieveLatestTermsOfServiceRequest;
use Google\Shopping\Merchant\Accounts\V1\TermsOfServiceKind;

/**
 * Demonstrates how to retrieve the latest TermsOfService.
 */
class RetrieveLatestTermsOfService
{
    /**
     * Retrieves the latest TermsOfService.
     *
     * @param array $config The configuration data.
     * @return void
     */
    public static function retrieveLatestTermsOfService($config): void
    {
        // Get OAuth credentials.
        $credentials = Authentication::useServiceAccountOrTokenFile();

        // Create client options.
        $options = ['credentials' => $credentials];

        // Create a TermsOfServiceServiceClient.
        $termsOfServiceServiceClient = new TermsOfServiceServiceClient($options);

        try {
            // Prepare the request.
            $request = new RetrieveLatestTermsOfServiceRequest([
                'region_code' => "US",
                'kind' => TermsOfServiceKind::MERCHANT_CENTER,
            ]);

            print "Sending Retrieve Latest TermsOfService request:\n";
            $response = $termsOfServiceServiceClient->retrieveLatestTermsOfService($request);

            print "Retrieved TermsOfService below\n";
            print $response->serializeToJsonString() . PHP_EOL;
        } catch (ApiException $e) {
            print $e->getMessage();
        }
    }

    /**
     * Helper to execute the sample.
     *
     * @return void
     */
    public function callSample(): void
    {
        $config = Config::generateConfig();

        self::retrieveLatestTermsOfService($config);
    }
}

// Run the script
$sample = new RetrieveLatestTermsOfService();
$sample->callSample();
// [END merchantapi_retrieve_latest_termsofservice]